hibernate-core/src/test/java/org/hibernate/orm/test/inheritance/discriminator/SingleTableNotNullDiscriminatorSuperClassTest.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.orm.test.inheritance.discriminator;
+
+import java.sql.Statement;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import jakarta.persistence.DiscriminatorValue;
+import jakarta.persistence.Entity;
+import jakarta.persistence.Id;
+import jakarta.persistence.Inheritance;
+import jakarta.persistence.InheritanceType;
+import jakarta.persistence.criteria.CriteriaQuery;
+import jakarta.persistence.criteria.Root;
+
+import org.hibernate.testing.TestForIssue;
+import org.hibernate.testing.orm.junit.DomainModel;
+import org.hibernate.testing.orm.junit.SessionFactory;
+import org.hibernate.testing.orm.junit.SessionFactoryScope;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+
+
+import static org.assertj.core.api.Assertions.assertThat;
+/**
+ * @author lrobot.qq@gmail.com
+ * how to run this test case: ./gradlew :hibernate-core:test --tests '*DiscriminatorSuperClassTest'
+ */
+@TestForIssue(jiraKey = "HHH-18282")
+@DomainModel(
+		annotatedClasses = {
+			SingleTableNotNullDiscriminatorSuperClassTest.RootEntity.class,
+			SingleTableNotNullDiscriminatorSuperClassTest.Val1Entity.class,
+			SingleTableNotNullDiscriminatorSuperClassTest.Val2Entity.class,
+			SingleTableNotNullDiscriminatorSuperClassTest.NotNullEntity.class
+		}
+)
+@SessionFactory
+public class SingleTableNotNullDiscriminatorSuperClassTest {
+
+	@AfterEach
+	public void tearDown(SessionFactoryScope scope) {
+		scope.inTransaction(
+				session ->
+						session.createQuery( "delete from root_ent" ).executeUpdate()
+		);
+	}
+
+	@Test
+	public void test(SessionFactoryScope scope) {
+		scope.inTransaction( session -> {
+			Val1Entity val1 = new Val1Entity();
+			val1.setId( 1L );
+
+			Val2Entity val2 = new Val2Entity();
+			val2.setId( 2L );
+
+			RootEntity root = new RootEntity();
+			root.setId( 3L );
+
+			session.persist( val1 );
+			session.persist( val2 );
+			session.persist( root );
+
+			session.doWork( connection -> {
+				try (Statement statement = connection.createStatement()) {
+					statement.executeUpdate(
+							"insert into root_ent (DTYPE, id) " +
+									"values ('other', 4)"
+					);
+				}
```

Comment:
There's no reason to resort to native SQL here. A HQL `insert` statement is shorter/more elegant.